controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1740,6 +1755,26 @@ func saveHostProvisioningSettings(host *metal3api.BareMetalHost, info *reconcile
 	return dirty, nil
 }
 
+func (r *BareMetalHostReconciler) saveHostFirmwareComponents(prov provisioner.Provisioner, info *reconcileInfo, hfc *metal3api.HostFirmwareComponents) (dirty bool, err error) {
+	dirty = false
+	if !reflect.DeepEqual(hfc.Status.Updates, hfc.Spec.Updates) {
+		hfc.Status.Updates = hfc.Spec.Updates
+
+		// Retrieve new information about the firmware components stored in ironic
+		components, err := prov.GetFirmwareComponents()
+		if err != nil {
+			info.log.Info("Failed to get new information for firmware components in ironic")
+			return dirty, err
+		}
+		hfc.Status.Components = components
+		t := metav1.Now()
+		hfc.Status.LastUpdated = &t
```

Comment:
Pretty sure the kube API handles this.

Reply:
ack, I will remove later to check this. Thanks!